internal/services/containerapps/helpers/container_apps.go

```diff
@@ -2259,27 +2262,25 @@ func SecretsDataSourceSchema() *pluginsdk.Schema {
 			Schema: map[string]*pluginsdk.Schema{
 				"identity": {
 					Type:        pluginsdk.TypeString,
-					Computed:    true,
-					Sensitive:   false,
+					Optional:    true,
 					Description: "The identity to use for accessing key vault reference.",
 				},
 
-				"key_vault_url": {
+				"key_vault_secret_id": {
 					Type:        pluginsdk.TypeString,
-					Computed:    true,
-					Sensitive:   true,
```

Comment:
fixing this and rest of data schema in next commit - made a couple of errors here

Reply:
fixed